Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1458,18 +1458,24 @@ private extension ChatChannelController {
     }
 
     private func getFirstUnreadMessageId() -> MessageId? {
-        let messages = self.messages.filter {
-            $0.type == .regular || $0.type == .reply
+        guard let currentUserRead = channel?.reads.first(where: {
+            $0.user.id == client.currentUserId
+        }) else {
+            return nil
```

Comment:
No, because if there are no reads from us, it means there are no unreads as well. We only must return `messages.last(where: { $0.type == .regular || $0.type == .reply })?.id` if `unreadCount` is more than zero. So we need to check first the `messagesUnreadCount` value

Reply:
It could also happen that no reads have yet been created -> no message read -> all of them are unread